ksqldb-parser/src/main/java/io/confluent/ksql/parser/SyntaxErrorValidator.java

```diff
@@ -127,16 +134,65 @@ public void syntaxError(
       return;
     }
 
-    if (offendingSymbol instanceof Token && isKeywordError(
-        message, ((Token) offendingSymbol).getText())) {
-      //Checks if the error is a reserved keyword error
-      final String tokenName = ((Token) offendingSymbol).getText();
-      final String newMessage =
-          "\"" + tokenName + "\" is a reserved keyword and it can't be used as an identifier."
-              + " You can use it as an identifier by escaping it as \'" + tokenName + "\' ";
-      throw new ParsingException(newMessage, e, line, charPositionInLine);
+    if (offendingSymbol instanceof Token) {
+      final StringBuilder sb = new StringBuilder();
+      sb.append(
+          String.format("Syntax Error\n"));
+      if (isKeywordError(message, ((Token) offendingSymbol).getText())) {
+        //Checks if the error is a reserved keyword error
+        final String tokenName = ((Token) offendingSymbol).getText();
+        sb.append(
+            String.format("\"%s\" is a reserved keyword and it can't be used as an identifier."
+                + " You can use it as an identifier by escaping it as '%s' ",
+                tokenName, tokenName));
+      } else if (e instanceof InputMismatchException) {
+        sb.append(
+            getExpectations(message,  e.getOffendingToken().getText(), line, charPositionInLine)
+        );
+      } else {
+        sb.append(message);
+      }
+      throw new ParsingException(sb.toString(), e, line, charPositionInLine);
     } else {
       throw new ParsingException(message, e, line, charPositionInLine);
     }
   }
+
+  private String getExpectations(
+      final String message,
+      final String offendingToken,
+      final int line,
+      final int charPositionInLine) {
+    final StringBuilder output = new StringBuilder();
+    final String expectingStr = message.split("expecting ")[1];
+    // If the command is mistyped, fins the most similar command and show it
+    if (line == 1 && charPositionInLine == 0) { // this is a command typo
+      output.append(String.format("Unknown statement '%s'\n",
+              offendingToken));
+      output.append(
+              String.format("Did you mean '%s'?", getMostSimilar(offendingToken)));
+    // In case of missing closing brackets or parentheses
+    } else if (message.contains("EOF")) {
+      output.append("Syntax error at or near \";\"\n");
```

Comment:
Would a statement that's missing a semicolon end up here? If it does, then the error message won't make sense.

Reply:
> Would a statement that's missing a semicolon end up here? If it does, then the error message won't make sense.

No. It does not.